File: 백준/Silver/10799. 쇠막대기/쇠막대기.cc
#include <bits/stdc++.h>
using namespace std;
string line;
int ret;
stack<char>s;
int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);

	cin >> line;

	for (int i = 0; i < line.length(); i++) {
		if (line[i] == '(')
			s.push(line[i]);
		else {
			s.pop();
			if (line[i - 1] == '(') ret += s.size();
			else ret++;
		}
	}
	cout << ret << '\n';
}

File: 백준/Silver/10799. 쇠막대기/쇠막대기.py
line = input()
s = []
ret = 0
for i in range(len(line)):
	if (line[i] == '('):
		s.append('(')
	else:
		s.pop()
		if (line[i - 1] == '('):
			ret += len(s)
		else: 
			ret += 1

print(ret)
